### PR TITLE
[WIP] BBCSPA-1288 drawer scroll top and scroll lock

### PR DESCRIPTION
**Pull request by Tuxosaurus**

### CODE REVIEW

File: src/drawer/index.tsx [hunk 1]
```diff
@@ -100,6 +104,24 @@ export default class Drawer extends PureComponent <DrawerProps, DrawerState> {
...
 
+  private scrollPosition: number = 0
+
+  scrollReset = () => {
+    // store current scroll position
+    this.scrollPosition = document.documentElement.scrollTop
+    // scroll to top
+    document.documentElement.scrollTop = 0
+    // block html scroll
+    document.querySelector('html').style.overflow = 'hidden'
```
Comment (wakooka) [inline src/drawer/index.tsx]: I thought you'd need to do it on both: `html` and `body`. Have you checked on multiple browsers?

File: src/drawer/index.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ export interface DrawerProps {
   readonly onClose?: () => void,
   readonly onChange?: (open: boolean) => void,
   readonly width?: string,
+  readonly height?: string,
```
[the remark this answers is not in the document]
Comment (Tuxosaurus) [reply src/drawer/index.tsx]: actually kairos already does it by adding a 'u-heightMax' class

**Comment by Tuxosaurus** [comment]
Ok so I checked the result on the SPA, and found an existing but that makes the current PR unusable: the drawer's closing animation is skipped when clicking the cross button in the topbar. Here that means that it never clears the "overflow: hidden" style.

Apparently the best thing would be to refactor the component to make use of react-transition-group.

(also, browser testing failed on Safari)

File: src/drawer/index.tsx [hunk 3]
```diff
@@ -100,6 +104,24 @@ export default class Drawer extends PureComponent <DrawerProps, DrawerState> {
...
+    document.documentElement.scrollTop = 0
+    // block html scroll
+    document.querySelector('html').style.overflow = 'hidden'
+  }
+
+  scrollRestore = () => {
+    // unblock html scroll
+    document.querySelector('html').style.overflow = 'visible'
+    // restore scroll position
+    document.documentElement.scrollTop = this.scrollPosition
```
[the remark this answers is not in the document]
Comment (loick) [reply src/drawer/index.tsx]: I agree!

File: src/drawer/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (pith) [reply src/drawer/index.tsx]: Be careful using vh on mobile can have unexpected behavior, for instance on Android the URL bar can be hidden so the vh change.

**Comment by pith** [comment]
Can you fix the conflicts before we review

**Comment by Tuxosaurus** [comment]
It's WIP for now, I will split it into 2 PRs, one for the scroll to top and the other for the scrollbar

**Pull request closed without merging**